pinot-controller/src/main/java/org/apache/pinot/controller/validation/OfflineSegmentIntervalChecker.java

```diff
@@ -81,18 +83,23 @@ private void validateOfflineSegmentPush(TableConfig tableConfig) {
     SegmentsValidationAndRetentionConfig validationConfig = tableConfig.getValidationConfig();
     if (numSegments >= 2 && StringUtils.isNotEmpty(validationConfig.getTimeColumnName())) {
       List<Interval> segmentIntervals = new ArrayList<>(numSegments);
-      List<String> segmentsWithInvalidInterval = new ArrayList<>();
+      List<String> segmentsWithInvalidInterval = new ArrayList<>(MAX_SEGMENTS_WITH_INVALID_INTERVALS_TO_LOG);
+      int numSegmentsWithInvalidIntervals = 0;
       for (OfflineSegmentZKMetadata offlineSegmentZKMetadata : offlineSegmentZKMetadataList) {
         Interval timeInterval = offlineSegmentZKMetadata.getTimeInterval();
         if (timeInterval != null && TimeUtils.timeValueInValidRange(timeInterval.getStartMillis()) && TimeUtils
             .timeValueInValidRange(timeInterval.getEndMillis())) {
           segmentIntervals.add(timeInterval);
         } else {
-          segmentsWithInvalidInterval.add(offlineSegmentZKMetadata.getSegmentName());
+          numSegmentsWithInvalidIntervals ++;
+          if (segmentsWithInvalidInterval.size() < MAX_SEGMENTS_WITH_INVALID_INTERVALS_TO_LOG) {
+            segmentsWithInvalidInterval.add(offlineSegmentZKMetadata.getSegmentName());
+          }
         }
       }
       if (!segmentsWithInvalidInterval.isEmpty()) {
-        LOGGER.warn("Table: {} has segments with invalid interval: {}", offlineTableName, segmentsWithInvalidInterval);
+        LOGGER.warn("Table: {} has {} segments with invalid interval. Listing up to {}: {}", offlineTableName,
```

Comment:
If there is an admin tool one can use to figure out the segments that have invalid interval, we can use that to find out the ones that are bad and just log the segment count here. (If one doesn't exist we can build one).

Reply:
hmm, that's a good idea. Periodically printing this is not really helping anyway. Will add that as a followup.